examples/molecular_dynamics/example_molecular_dynamics.cpp

```diff
@@ -0,0 +1,178 @@
+{
+  using memory_space = MemorySpace;
+  using size_type = std::size_t;
+  static KOKKOS_FUNCTION size_type size(Neighbors<MemorySpace> const &x)
+  {
+    return x.particles_.extent(0);
+  }
+  static KOKKOS_FUNCTION auto get(Neighbors<MemorySpace> const &x, size_type i)
+  {
+    return attach(intersects(Sphere{x.particles_(i), x.cutoff_radius_}),
+                  (int)i);
+  }
+};
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = ExecutionSpace::memory_space;
+
+  ExecutionSpace execution_space{};
+
+  float const dx = 1.f;
+  float const dy = 1.f;
+  float const dz = 1.f;
+  int const nx = 100;
+  int const ny = 100;
+  int const nz = 100;
+  int const n = 4 * nx * ny * nz;
+
+  float const r = 1.f; // cut-off radius
+
+  Kokkos::Profiling::pushRegion("setup");
+  Kokkos::View<ArborX::Point *, MemorySpace> particles(
+      Kokkos::view_alloc("points", Kokkos::WithoutInitializing), n);
+  Kokkos::parallel_for(
+      "make_particles",
+      Kokkos::MDRangePolicy<Kokkos::Rank<3>, ExecutionSpace>(
+          execution_space, {0, 0, 0}, {nx, ny, nz}),
+      KOKKOS_LAMBDA(int i, int j, int k) {
+        int const id = i * ny * nz + j * nz + k;
+        particles[4 * id + 0] = {i * dx + .0f, j * dy + .0f, k * dz + .0f};
+        particles[4 * id + 1] = {i * dx + .0f, j * dy + .5f, k * dz + .5f};
+        particles[4 * id + 2] = {i * dx + .5f, j * dy + .0f, k * dz + .5f};
+        particles[4 * id + 3] = {i * dx + .5f, j * dy + .5f, k * dz + .0f};
+      });
+
+  Kokkos::View<float **, MemorySpace> velocities(
+      Kokkos::view_alloc("velocities", Kokkos::WithoutInitializing), n, 3);
+  { // scope so that random number generation resources are released
+    using RandomPool = Kokkos::Random_XorShift64_Pool<ExecutionSpace>;
+    RandomPool random_pool(5374857);
+
+    Kokkos::parallel_for(
+        "assign_velocities",
+        Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+        KOKKOS_LAMBDA(int i) {
+          RandomPool::generator_type generator = random_pool.get_state();
+          velocities(i, 0) = generator.frand(0.f, 1.f);
+          velocities(i, 1) = generator.frand(0.f, 1.f);
+          velocities(i, 2) = generator.frand(0.f, 1.f);
+          random_pool.free_state(generator);
+        });
+  }
+  // TODO scale velocities
+  Kokkos::Profiling::popRegion();
+
+  ArborX::BVH<MemorySpace> index(execution_space, particles);
+
+  Kokkos::View<int *, MemorySpace> indices("indices", 0);
+  Kokkos::View<int *, MemorySpace> offsets("offsets", 0);
+  index.query(execution_space, Neighbors<MemorySpace>{particles, r}, indices,
+              offsets);
+
+  Kokkos::View<float * [3], MemorySpace> forces("forces", n);
+  Kokkos::parallel_for(
+      "compute_forces",
+      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+      KOKKOS_LAMBDA(int i) {
+        auto const x_i = particles(i)[0];
```

Comment:
Why not use `for (int d = 0; d < 3; ++d)` in the loop? Most operates are triplicated. This would results in a shorter code.

Reply:
Won't fix.  This is how Cabana and Lammps write their code.